### PR TITLE
Add long-range fork consensus rule

### PR DESCRIPTION
**Pull request by xqft**

### CODE REVIEW

File: operator/mina/lib/src/consensus_state.rs [hunk 1]
```diff
@@ -1,47 +1,174 @@
...
+
+    min(
+        candidate.min_window_density.as_u32(),
+        projected_window_density,
+    )
 }
 
-fn hash_last_vrf(chain: &MinaStateProtocolStateValueStableV2) -> String {
+fn relative_sub_window(state: &MinaConsensusState) -> u32 {
+    (state.global_slot() / SLOTS_PER_SUB_WINDOW) % SUB_WINDOWS_PER_WINDOW
```
Comment (gabrielbosio) [inline operator/mina/lib/src/consensus_state.rs:127]: We can use `state.body.constants.slots_per_sub_window` instead of `SLOTS_PER_SUB_WINDOW`.

Comment (xqft) [reply operator/mina/lib/src/consensus_state.rs:127]: problem here is that the parameter is a consensus state, which doesn't have the protocol constants. Would it be better if I hardcode all the constants and check if they match with the states?

Comment (gabrielbosio) [reply operator/mina/lib/src/consensus_state.rs:127]: Ok, it makes sense for now

Comment (gabrielbosio) [reply operator/mina/lib/src/consensus_state.rs:127]: Leaving like this for now

**Review by gabrielbosio — APPROVED** [review]

**Pull request merged**